### PR TITLE
Bug fix/search 317

### PR DESCRIPTION
**Pull request by Dronplane**
### Scope & Purpose

Fix available fields detection.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/971
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/IResearch/IResearchInvertedIndexMeta.h [hunk 1]
```diff
@@ -184,6 +191,31 @@ struct InvertedIndexField {
...
 
+struct IResearchInvertedIndexMeta;
+
+struct IResearchInvertedIndexMetaIndexingContext {
+  IResearchInvertedIndexMetaIndexingContext(
+      IResearchInvertedIndexMeta const* field, bool add = true);
+
+  void addField(InvertedIndexField const& field);
+
+  absl::flat_hash_map<std::string_view,
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.h:202]:
```suggestion
  containers::FlatHashMap<std::string_view,
```

File: arangod/IResearch/IResearchInvertedIndexMeta.cpp [hunk 2]
```diff
@@ -66,6 +66,58 @@ constexpr std::string_view kStoredValuesFieldName = "storedValues";
...
+  map[self].emplace(field.path());
+  arangodb::iresearch::kludge::mangleNested(nestedObjectsPath);
+  for (auto const& sf : field._fields) {
+    if (!sf._fields.empty()) {
+      gatherNestedNulls(field.path(), map, sf);
+    }
+    map[nestedObjectsPath].emplace(sf.path());
+  }
+}
+#endif
```
Comment (MBkkt) [inline arangod/IResearch/IResearchInvertedIndexMeta.cpp:90]: Move definition to ee?

**Review by MBkkt — CHANGES_REQUESTED** [review]

**Review by MBkkt — APPROVED** [review]

File: arangod/IResearch/AqlHelper.cpp [hunk 3]
```diff
@@ -62,8 +62,19 @@ arangodb::aql::AstNodeType const CmpMap[]{
...
 
+auto getNested(
+    std::string_view parent,
+    std::span<const arangodb::iresearch::InvertedIndexField> fields) noexcept {
+  if (parent.ends_with("[*]")) {
+    parent = parent.substr(0, parent.size() - 3);
+  }
+  return std::find_if(
+      std::begin(fields), std::end(fields),
+      [parent](const auto& field) noexcept { return field.path() == parent; });
```
[the remark this answers is not in the document]
Comment (gnusi) [reply arangod/IResearch/AqlHelper.cpp:73]: EE?

File: arangod/IResearch/AqlHelper.h [hunk 4]
```diff
@@ -535,8 +536,10 @@ bool attributeAccessEqual(aql::AstNode const* lhs, aql::AstNode const* rhs,
 /// @brief generates field name from the specified 'node'
 /// @returns true on success, false otherwise
 ////////////////////////////////////////////////////////////////////////////////
-bool nameFromAttributeAccess(std::string& name, aql::AstNode const& node,
-                             QueryContext const& ctx);
+bool nameFromAttributeAccess(
+    std::string& name, aql::AstNode const& node, QueryContext const& ctx,
+    bool filter, std::span<InvertedIndexField const> fields,
+    std::span<InvertedIndexField const>* subFields = nullptr);
```
Comment (gnusi) [inline arangod/IResearch/AqlHelper.h:542]: can't we pass `std::span` by value?

**Pull request merged**